pkg/network/tracer/tracer.go

```diff
@@ -139,12 +141,17 @@ func newTracer(cfg *config.Config, telemetryComponent telemetryComponent.Compone
 	}
 
 	if cfg.ServiceMonitoringEnabled {
-		if !usmconfig.IsUSMSupported() {
-			errStr := fmt.Sprintf("Universal Service Monitoring (USM) requires a Linux kernel version of %s or higher. We detected %s", usmconfig.MinimumKernelVersion, currKernelVersion)
+		if err := usmconfig.CheckUSMSupported(cfg); err != nil {
+			if !errors.Is(err, usmconfig.ErrNotSupported) {
+				return nil, err
+			}
+
 			if !cfg.NPMEnabled {
-				return nil, fmt.Errorf(errStr)
+				return nil, err
 			}
```

Comment:
if npm is disabled, but ebpless is enabled - usm won't run.
please disable ebpfless if npm is disabled (adjust_npm)

Reply:
This code won't run if npm is disabled, so I am not sure what the concern is here.

Reply:
Actually I do see that we can end up here even if npm is disabled. This is strange, why do we end up here when usm is enabled and npm is disabled?

Reply:
This is because of NPM being enabled implicitly if USM is enabled. I am going to add a comment to that line so that is clear why we end up there.

Reply:
Did we reach a conclusion? I imagine this isn't the first time we've had two config values be incompatible and for the agent to need to pick a 'winner'.